core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,90 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import jenkins.model.Messages;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ */
+public abstract class ItemCategory implements ModelObject, ExtensionPoint {
+
+    public static int MIN_WEIGHT = Integer.MIN_VALUE;
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Identifier, e.g. "category-id-default", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The icon class specification e.g. 'category-icon-folder', 'category-icon-default', etc.
+     *
+     * @return the icon class specification
+     */
+    public abstract String getIconClassName();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * Helpful to set the order.
+     *
+     * @return the weight
+     */
+    public abstract int getWeight();
```

Comment:
Ahh .... I see now what `weight` is :smile:. Why not just name it "order" ala the method description here?